security/webauthn/src/test/java/AbstractWebAuthnTest.java

```diff
@@ -0,0 +1,156 @@
+import static io.restassured.RestAssured.given;
+
+import java.util.function.Consumer;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.quarkus.security.webauthn.WebAuthnController;
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.security.webauthn.WebAuthnEndpointHelper;
+import io.quarkus.test.security.webauthn.WebAuthnHardware;
+import io.restassured.RestAssured;
+import io.restassured.filter.Filter;
+import io.restassured.specification.RequestSpecification;
+import io.vertx.core.json.JsonObject;
+
+public abstract class AbstractWebAuthnTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("AbstractWebAuthnTest");
+
+    protected abstract RestService getApp();
+
+    enum User {
+        USER;
+    }
+
+    enum Endpoint {
+        DEFAULT,
+        MANUAL;
+    }
+
+    @Test
+    public void testWebAuthnUser() {
+        testWebAuthn("Roosvelt", User.USER, Endpoint.MANUAL);
+    }
+
+    private void testWebAuthn(String userName, User user, Endpoint endpoint) {
+        Filter cookieFilter = new CookieFilter();
+        WebAuthnHardware token = new WebAuthnHardware();
+
+        verifyLoggedOut(cookieFilter);
+
+        // two-step registration
+        String challenge = WebAuthnEndpointHelper.invokeRegistration(userName, cookieFilter);
+        JsonObject registrationJson = token.makeRegistrationJson(challenge);
+        if (endpoint == Endpoint.DEFAULT)
+            WebAuthnEndpointHelper.invokeCallback(registrationJson, cookieFilter);
+        else {
+            invokeCustomEndpoint("/register", cookieFilter, request -> {
+                WebAuthnEndpointHelper.addWebAuthnRegistrationFormParameters(request, registrationJson);
```

Comment:
What is happening when you are using `WebAuthHardware` and `WebAuthEndpointHelper` is that exactly same configuration of headers, encoding, signature, keys generator is used as is already tested by the extension unit tests and quickstart. 

I think the proper way to do this is to actually configure Microsoft Playwright we use for UI testing to provide hardware token. I think it can be difficult and to stop using this utility classes would increase time required for TD, so I'll let final decision on you.

This just seems very much like mocking.

Reply:
Ok, at the beginning of this task, I was thinking of doing some selenium UI interaction with Webdriver but firstly I wanted to have something working as we already have.
But, if you use Microsoft Playwright I for UI testing will use it, do you know some uses case in quarkus test suite? I've checked that it is in the Quarkus test framework (DevModeQuarkusService) but that's all.

Reply:
> I've checked that it is in the Quarkus test framework (DevModeQuarkusService) but that's all.

Yeah, that's what I meant, if you can make it work with `org.htmlunit`, fine with me. Uou can just fulltext search for it inside QE Test FW, but I warn you, this is probably not going to be easy :-)

Reply:
You were right is not easy, I've tried with playwright (https://github.com/microsoft/playwright/issues/7276) and with webclient (htmlunit) without success, however, I've found something that could be interesting with selenium webdriver-->
https://www.selenium.dev/documentation/webdriver/interactions/virtual_authenticator/
The register of a user seems to work but not really, I will do a push, could you take a look?

Reply:
> I will do a push, could you take a look? 

I can't really see anything or test it, CI fails over _[ERROR] /home/runner/work/quarkus-test-suite/quarkus-test-suite/security/webauthn/src/test/java/AbstractWebAuthnSelenium.java:[42,9] cannot find symbol_

Tests with selenium doesn't look bad, but it looks very similar to HtmlUnit and Playwright. I'd suggest that if you are in hurry as you say, let's start simple without it and do second PR. It will be easier to manage. Decision is completely yours!

Reply:
Also it brings in additional dependencies that needs to be managed, so there is cost of running CI.

Reply:
> > I will do a push, could you take a look?
> 
> I can't really see anything or test it, CI fails over _[ERROR] /home/runner/work/quarkus-test-suite/quarkus-test-suite/security/webauthn/src/test/java/AbstractWebAuthnSelenium.java:[42,9] cannot find symbol_
>   _Fixed it! Can you try now?_
> Tests with selenium doesn't look bad, but it looks very similar to HtmlUnit and Playwright. I'd suggest that if you are in hurry as you say, let's start simple without it and do second PR. It will be easier to manage. Decision is completely yours!
let's do a try to this and if tomorrow morning still not see the light then I will do simplest as possible.